clang/test/SemaSYCL/sycl-varargs-cconv.cpp

```diff
@@ -0,0 +1,4 @@
+// RUN: %clang_cc1 -fsycl-is-device -verify -fsyntax-only -x c++ -emit-llvm-only %s
+// expected-no-diagnostics
+
+int __cdecl foo(int, ...);
```

Comment:
Please, add test case with the `foo` call from the SYCL kernel and check that compiler emits an error.

Reply:
May I do that as a separate patch or do you want that along with this one?